anchore_engine/services/policy_engine/engine/feeds/grypedb_sync.py

```diff
@@ -73,41 +73,82 @@ class GrypeDBSyncManager:
         TooManyActiveGrypeDBs if more than one
 
         return: Active GrypeDBFeedMetadata
         rtype: GrypeDBFeedMetadata
         """
-        try:
-            active_grypedb = cls._query_active_dbs()
-        except sqlalchemy.orm.exc.MultipleResultsFound:
+        active_grypedbs = cls._query_active_dbs(session)
+
+        if len(active_grypedbs) > 1:
             logger.error("Too many active grype dbs found in db")
             raise TooManyActiveGrypeDBs
 
-        if not active_grypedb:
+        if len(active_grypedbs) == 0:
             logger.error("No active grype db found in the database")
             raise NoActiveGrypeDB
 
-        return active_grypedb
+        return active_grypedbs[0]
 
     @classmethod
-    def _query_active_dbs(cls) -> Optional[GrypeDBFeedMetadata]:
+    def _query_active_dbs(cls, session: Session) -> List[GrypeDBFeedMetadata]:
         """
-        Runs query against db to get active dbs. Uses one_or_none so raises error if more than one active db
+        Runs query against db to get active dbs and returns them in array
 
-        return: Instance of GrypeDBFeedMetadata or None
-        rtype: GrypeDBFeedMetadata
+        return: List of active GrypeDBFeedMetadata
+        rtype: List[GrypeDBFeedMetadata]
         """
-        db = get_thread_scoped_session()
         return (
-            db.query(GrypeDBFeedMetadata)
+            session.query(GrypeDBFeedMetadata)
             .filter(GrypeDBFeedMetadata.active == True)
-            .one_or_none()
+            .all()
         )
 
+    @classmethod
+    def resolve_and_get_active_grypedb(cls, session: Session) -> GrypeDBFeedMetadata:
+        """
+        Public function to retrieve the current active grypedb in the database.
+        If more than one present it calls another function that resolves it by choosing the most recent one
+        *Note that this function may make commits on session object if there is more than one active db
+        """
+        try:
+            return cls._get_active_grypedb(session)
+        except TooManyActiveGrypeDBs:
+            logger.info(
+                "Too many active grype dbs found in the db. Resolving by selecting most recently created"
+            )
+            cls._resolve_active_db(session)
+            return cls._get_active_grypedb(session)
+
+    @classmethod
+    def _resolve_active_db(cls, session: Session) -> None:
+        """
+        Resolves active db by querying all active ones sorted by created date.
+        Sets all older ones to inactive
+        Commits to session object
+        """
+        # get all active dbs and select the most recent one to be the correct active db
+        active_dbs = (
+            session.query(GrypeDBFeedMetadata)
+            .filter(GrypeDBFeedMetadata.active == True)
+            .order_by(GrypeDBFeedMetadata.created_at.desc())
+            .all()
+        )
+        resolved_active = active_dbs.pop(0)
```

Comment:
This will fail when active the list is empty. I understand that might not be the case when this funciton is called but a slightly safer and efficient way of doing this is in python is list slicing  
```
if active_dbs and len(active_dbs) > 1:
  for outdated_db in active_dbs[1:]:
    outdated_db.active = False
```